NoHttpResponseException: 127.0.0.1:3100 failed to respond

Error occurs periodically when using Apache HTTP Client:

```
10:07:35,277 |-ERROR in com.github.loki4j.logback.LokiApacheHttpAppender[LOKI] - Error while sending Batch #cd4aa90f29e17 (108 records) to Loki java.util.concurrent.CompletionException: java.lang.RuntimeException: Error while sending batch to Loki
	at java.util.concurrent.CompletionException: java.lang.RuntimeException: Error while sending batch to Loki
	at 	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at 	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1702)
	at 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at 	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.RuntimeException: Error while sending batch to Loki
	at 	at com.github.loki4j.logback.LokiApacheHttpAppender.lambda$sendAsync$1(LokiApacheHttpAppender.java:80)
	at 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at 	... 3 common frames omitted
Caused by: org.apache.http.NoHttpResponseException: 127.0.0.1:3100 failed to respond
	at 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:141)
	at 	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
	at 	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
	at 	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
	at 	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:157)
	at 	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
	at 	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
	at 	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
	at 	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at 	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at 	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at 	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at 	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at 	at com.github.loki4j.logback.LokiApacheHttpAppender.lambda$sendAsync$1(LokiApacheHttpAppender.java:74)
	at 	... 4 common frames omitted
```

--- a/docs/docus/docs/appenders.md
+++ b/docs/docus/docs/appenders.md
@@ -121,7 +121,8 @@ However, there are some client-specific settings with their default values:
     ...
     <!-- Maximum number of HTTP connections setting for HttpClient -->
     <maxConnections>1</maxConnections>
-    <!-- Maximum life span of persistent connections for HttpClient -->
-    <connectionTtlMs>30000</connectionTtlMs>
+    <!-- A duration of time which the connection can be safely kept idle for later reuse. -->
+    <!-- This value can not be greater than `server.http-idle-timeout` in your Loki config -->
+    <connectionKeepAliveMs>120000</connectionKeepAliveMs>
 </appender>
 ```

--- a/src/main/java/com/github/loki4j/logback/LokiApacheHttpAppender.java
+++ b/src/main/java/com/github/loki4j/logback/LokiApacheHttpAppender.java
@@ -2,15 +2,17 @@ package com.github.loki4j.logback;
 
 import java.io.IOException;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 
+import org.apache.http.HttpResponse;
 import org.apache.http.client.config.RequestConfig;
 import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ConnectionKeepAliveStrategy;
 import org.apache.http.entity.ByteArrayEntity;
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.impl.client.HttpClients;
 import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.protocol.HttpContext;
 import org.apache.http.util.EntityUtils;
 
 /**
@@ -24,22 +26,30 @@ public class LokiApacheHttpAppender extends AbstractLoki4jAppender {
     private int maxConnections = 1;
 
     /**
-     * Maximum life span of persistent connections for HttpClient
+     * A duration of time which the connection can be safely kept
+     * idle for later reuse. This value can not be  greater than
+     * server.http-idle-timeout in your Loki config
      */
-    private long connectionTtlMs = 30_000;
+    private long connectionKeepAliveMs = 120_000;
 
     private CloseableHttpClient client;
     private Function<byte[], HttpPost> requestBuilder;
 
     @Override
     protected void startHttp(String contentType) {
-        var cm = new PoolingHttpClientConnectionManager(connectionTtlMs, TimeUnit.MILLISECONDS);
+        var cm = new PoolingHttpClientConnectionManager();
         cm.setMaxTotal(maxConnections);
         cm.setDefaultMaxPerRoute(maxConnections);
 
         client = HttpClients
             .custom()
             .setConnectionManager(cm)
+            .setKeepAliveStrategy(new ConnectionKeepAliveStrategy(){
+                @Override
+                public long getKeepAliveDuration(HttpResponse response, HttpContext context) {
+                    return connectionKeepAliveMs;
+                }
+            })
             .setDefaultRequestConfig(RequestConfig
                 .custom()
                 .setSocketTimeout((int)connectionTimeoutMs)
@@ -85,8 +95,8 @@ public class LokiApacheHttpAppender extends AbstractLoki4jAppender {
         this.maxConnections = maxConnections;
     }
 
-    public void setConnectionTtlMs(long connectionTtlMs) {
-        this.connectionTtlMs = connectionTtlMs;
+    public void setConnectionKeepAliveMs(long connectionKeepAliveMs) {
+        this.connectionKeepAliveMs = connectionKeepAliveMs;
     }
 
 }

